modules/core/keeper/msg_server.go

```diff
@@ -556,9 +564,18 @@ func (k Keeper) Timeout(goCtx context.Context, msg *channeltypes.MsgTimeout) (*c
 	}
 
 	// Perform TAO verification
-	if err := k.ChannelKeeper.TimeoutPacket(ctx, msg.Packet, msg.ProofUnreceived, msg.ProofHeight, msg.NextSequenceRecv); err != nil {
+	// If the timeout was already received we want to perform a no-op. We use a cached context to prevent
+	// accidental state changes
+	cacheCtx, writeFn := ctx.CacheContext()
+	if err := k.ChannelKeeper.TimeoutPacket(cacheCtx, msg.Packet, msg.ProofUnreceived, msg.ProofHeight, msg.NextSequenceRecv); err != nil {
+		// timeout already received
+		// NOTE: ordered channels will error on a closed channel upon replay
```

Comment:
I need to give some thought about considering closed channels as redundant relaying